pallets/amm/src/lib.rs

```diff
@@ -399,7 +401,7 @@ decl_module! {
 
 			Self::deposit_event(RawEvent::RemoveLiquidity(who.clone(), asset_a, asset_b, liquidity_amount));
 
-			if liquidity_left == 0 {
+			if liquidity_left < MIN_POOL_LIQUIDITY_LIMIT {
```

Comment:
Add also a test for this scenario where liquidity left is > 0 but less < the limit.

Reply:
We also need to do something with the remaining tokens. We shouldn't just leave them sitting there I guess?

Reply:
Agree.
Would be grateful for suggestions.

Reply:
I would check if when exiting pool you left more than minimal pool amount or exactly 0. We'd make sure you can't leave less in the pool and therefore this will never happen (would need comment/doc tho)

Reply:
* of your tokens.

Reply:
> I would check if when exiting pool you left more than minimal pool amount or exactly 0. We'd make sure you can't leave less in the pool and therefore this will never happen (would need comment/doc tho)

If I understand correctly:

- if liquidity_left >= MIN_POOL_LIQ_LIMIT: **remove liquidity**
- if liquidity_left == 0: **remove liquidity & destroy pool**
- if liquidity_left < MIN_POOL_LIQ_LIMIT: **report Error to User?**

Reply:
Yes! 
but 1 and 2 are pool_liquidity_left 
3 is users_outstanding_liquidity_left (share tokens) < MIN_POOL_LIQ_LIMIT

Reply:
Because If we only checked pool_liquidity_left in all cases, we could arrive to a state where there are two liq providers that combined have more than MIN_POOL_LIQ_LIMIT but can't withdraw liquidity because their own liquidity left is smaller than < MIN_POOL_LIQ_LIMIT.